platforms/web/lib/conversion.ts

```diff
@@ -88,3 +77,106 @@ export async function plainToRich(plainText: string, inMessageFormat: boolean) {
+      subsequent typing will replace the final `\n` to give `word\nanother word`
+  - text with <div> separation if `cmd + enter to send` is enabled and `enter` is used to insert
+    the linebreak
+    - in this case, inserting a newline inserts `<div><br></div>`, and then subsequent typing 
+      replaces the <br> tag with the new content
+  - mentions (ie <a> tags with special attributes) which can be at the top level, or nested inside
+    a div 
+What we need to do is to get this input into a good shape for the markdown parser in the rust model. 
+Because of some of the intricacies of how text content is parsed when you use `.innerHTML` vs `.innerText`
+we do it manually so that we can extract:
+  - text content from any text nodes exactly as the user has written it, so that there is no escaping
+    of html entities like < or &
+  - mentions in their pure html form so that they can be passed through as valid html, as the mentions
+    in the plain text composer can be parsed into mentions inside the rust model
+*/
+export function plainTextInnerHtmlToMarkdown(innerHtml: string): string {
+    // Parse the innerHtml into a DOM and treat the `body` as the `composer`
+    const { body: composer } = new DOMParser().parseFromString(
+        innerHtml,
+        'text/html',
+    );
+
+    // Create an iterator to allow us to traverse the DOM node by node, excluding the
+    // text nodes inside mentions
+    const iterator = document.createNodeIterator(
+        composer,
+        undefined,
+        nodeFilter,
+    );
+    let node = iterator.nextNode();
+
+    // Use this to store the manually built markdown output
+    let markdownOutput = '';
+
+    while (node !== null) {
+        // TEXT NODES - `node` represents the text node
+        const isTextNode = node.nodeName === '#text';
+
+        // MENTION NODES - `node` represents the enclosing <a> tag
+        const isMention = node.nodeName === 'A';
+
+        // LINEBREAK DIVS - `node` represents the enclosing <div> tag
+        const isDivContainingBreak =
+            node.nodeName === 'DIV' &&
+            node.childNodes.length === 1 &&
+            node.firstChild?.nodeName === 'BR';
+
+        if (isDivContainingBreak) {
+            markdownOutput += NEWLINE_CHAR;
+        } else if (isTextNode) {
+            // content is the text itself, unescaped i.e. > is >, not &gt;
+            let content = node.textContent;
+            if (shouldAddNewlineCharacter(node)) {
+                content += NEWLINE_CHAR;
+            }
+            markdownOutput += content;
+        } else if (isMention) {
+            // content is the html of the mention i.e. <a ...attributes>text</a>
+            let content = node.firstChild?.parentElement?.outerHTML ?? '';
+            if (shouldAddNewlineCharacter(node)) {
+                content += NEWLINE_CHAR;
+            }
+            markdownOutput += content;
+        }
+
+        node = iterator.nextNode();
+    }
```

Comment:
A bit scared about this code, if we step upon nodes that are not expected here, they will just get ignored IIUC. 
Does it make sense to, at least, log nodes getting ignored ? (at least on debug if that affects performance)

Reply:
Seems fine with the logging. 
I guess we'll see if that ever triggers.